Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -4096,11 +4096,25 @@ Ref<HitTestingTransformState> RenderLayer::createLocalTransformState(RenderLayer
     return transformState.releaseNonNull();
 }
 
+static RefPtr<Element> flattenedParent(Element* element)
+{
+    if (!element)
+        return nullptr;
+    RefPtr parent = element->parentElementInComposedTree();
+    while (parent) {
+        if (parent->computedStyle()->display() != DisplayType::Contents)
+            break;
+        parent = parent->parentElementInComposedTree();
+    }
+    return parent;
+}
+
 bool RenderLayer::ancestorLayerIsDOMParent(const RenderLayer* ancestor) const
 {
     if (!ancestor)
         return false;
-    if (renderer().element() && ancestor->renderer().element() == renderer().element()->parentElementInComposedTree())
+    auto parent = flattenedParent(renderer().element());
+    if (parent && ancestor->renderer().element() == parent)
```

Comment:
I guess `flattenedParent(renderer().element()) == ancestor->renderer().element()` would be sufficient but ofc checking against parent is less error prone.

Reply:
(totally not suggesting to change it:)

Reply:
I guess I was worried about `nullptr == nullptr`, not sure how probable that actually is.

Reply:
yeah, probably close to 0, but then again being defensive here is not a bad thing.